compiler/rustc_borrowck/src/diagnostics/conflict_errors.rs

```diff
@@ -498,21 +500,98 @@ impl<'cx, 'tcx> MirBorrowckCtxt<'cx, 'tcx> {
                     self.suggest_cloning(err, ty, expr, None);
                 }
             }
+
+            self.suggest_ref_for_dbg_args(expr, span, move_span, err);
+
             if let Some(pat) = finder.pat {
-                *in_pattern = true;
-                let mut sugg = vec![(pat.span.shrink_to_lo(), "ref ".to_string())];
-                if let Some(pat) = finder.parent_pat {
-                    sugg.insert(0, (pat.span.shrink_to_lo(), "ref ".to_string()));
-                }
-                err.multipart_suggestion_verbose(
-                    "borrow this binding in the pattern to avoid moving the value",
-                    sugg,
-                    Applicability::MachineApplicable,
+                // FIXME: any better way to check this?
+                let from_std = self.infcx.tcx.sess.opts.real_rust_source_base_dir.clone().map_or(
+                    false,
+                    |root| {
+                        let file_path =
+                            match self.infcx.tcx.sess.source_map().span_to_filename(move_span) {
+                                FileName::Real(name) => {
+                                    name.clone().into_local_path().unwrap_or_default()
+                                }
+                                other => PathBuf::from(other.prefer_local().to_string()),
+                            };
+                        file_path.starts_with(&root.join("library/std/"))
+                    },
                 );
+                // it's useless to suggest inserting `ref` when the span comes from std library
+                // anyway, user can not modify std library in most cases, so let's keep it quite?
+                if !from_std {
```

Comment:
I think you'll want to forbid the suggestion whenever it points anywhere but to local code, the standard library is just the most common external crate in our test suite, but the issue probably exists for any dependency?

Reply:
changed it to:

```rust
    && !move_span.is_dummy()
    && !self.infcx.tcx.sess.source_map().is_imported(move_span)
```